### DISCUSSION TITLE
AndroidSdk--请问PAGShapeLayer是否支持着色

### DISCUSSION THREAD

**Discussion by carleft** [Tencent/libpag#1932] [Q&A] [answered]
目前的需求是，动画的颜色按照服务端下发展示

以前的lottie动画有如下方法可以着色：
    /**
     * lottie animation 改色
     */
    public static void setLottieColor(LottieAnimationView view, int color) {
        KeyPath keyPath = new KeyPath("**");
        LottieValueCallback<ColorFilter> callback = new LottieValueCallback<>(new PorterDuffColorFilter(color, PorterDuff.Mode.SRC_ATOP));
        view.addValueCallback(keyPath, LottieProperty.COLOR_FILTER, callback);
    }

现在改为pag后，所有图层都是ShapeLayer，是否有着色功能？
能否将shapeLayer中的shape导出为shapeDrawble然后设置颜色

Android Sdk版本
4.3.33
示例动画文件，该动画文件由多个shape图层构成
[collection_add_favor.zip](https://github.com/Tencent/libpag/files/13497506/collection_add_favor.zip)

**Comment by kevingpqi123** [comment, marked as answer]
PAGShapeLayer 不支持外部设置颜色，但实色图层（PAGSolidLayer）支持

**Comment by carleft** [comment]
1. 有没有考虑开发这种功能
2. 有没有什么好的替代方案，求指点

**Reply by kevingpqi123** [reply]
暂时没有这个计划，可以看下是否可以通过实色图层 + 蒙版实现

**Reply by domchen** [reply]
你需要的修改复杂 Shape 图层颜色的功能其实用 Solid Layer 的接口就可以实现。具体是设计师那边把复杂图形作为 mask 去擦除你的纯色图层就可以了，你再改纯色图层的颜色就可以起到给复杂图形统一换颜色的功能。其他业务也都是这么实现的。
